internal/auth/service_list_auth_methods_refresh.go

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package auth
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/refreshtoken"
+)
+
+func (s *AuthMethodService) ListAuthMethodsRefresh(
+	ctx context.Context,
+	tok *refreshtoken.Token,
+	scopeIds []string,
+	withUnauthenticatedUser bool,
+	pageSize int,
+	grantsHash []byte,
+	filterItemFn func(AuthMethod) (bool, error),
+) (*ListAuthMethodsResponse, error) {
+	const op = "auth.(*AuthMethodService).ListAuthMethodsRefresh"
+
+	// Request and combine deleted ids from the DB for ldap, oidc, and passwords.
+	var deletedIds []string
+	var transactionTimestamp time.Time
+	if _, err := s.writer.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{}, func(r db.Reader, w db.Writer) error {
+		deletedLdapIds, err := s.ldapRepo.ListDeletedAuthMethodIds(ctx, tok.UpdatedTime, WithReaderWriter(ctx, r, w))
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		deletedOidcIds, err := s.oidcRepo.ListDeletedAuthMethodIds(ctx, tok.UpdatedTime, WithReaderWriter(ctx, r, w))
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		deletedPwIds, err := s.passwordRepo.ListDeletedAuthMethodIds(ctx, tok.UpdatedTime, WithReaderWriter(ctx, r, w))
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		transactionTimestamp, err = r.Now(ctx)
+		if err != nil {
+			return errors.Wrap(ctx, err, op)
+		}
+		deletedIds = append(deletedLdapIds, append(deletedOidcIds, deletedPwIds...)...)
+		return nil
+	}); err != nil {
```

Comment:
This logic is what requires us to create a service struct. We need the deleted auth methods from all 3 domains to be collated in a single transaction with a single transaction timestamp. This requires a DB reader, which isn't available in the handlers, so we need to create the service struct in the controller. If we didn't need to do this all in a single transaction, we could forgo the whole struct.